back/transcendence_django/back_user/views.py

```diff
@@ -165,3 +166,51 @@ def get_game_summaries(request) -> JsonResponse:
         return JsonResponse(
             {"error": "Invalid request data: " + str(e)}, status=HTTPStatus.BAD_REQUEST
         )
+
+
+@method_decorator(csrf_protect, name="dispatch")
+class UpdateUsernameView(APIView):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.username: str = ""
+        self.user_id: str = ""
+
+    def write_username_in_db(self):
+        try:
+            user = CustomUser.objects.get(id=self.user_id)
+            user.username = self.username
+            user.save()
+            return user
+        except CustomUser.DoesNotExist:
+            logger.error("User with the provided ID does not exist.")
+            return None
+
+    def username_already_taken(self) -> bool:
+        if CustomUser.objects.filter(username=self.username).exists():
+            logger.info("Username is already taken")
+            return True
+        return False
+
+    def update_username(self):
+        if self.username_already_taken():
+            return None, Response({"error": "Username already taken."}, status=400)
+
+        user = self.write_username_in_db()
```

Comment:
add the user as an attribute member to avoid write_username_in_db returns the user.